internal/experiment/echcheck/handshake.go

```diff
@@ -93,14 +84,18 @@ func handshakeMaybePrintWithECH(doprint bool) string {
 	return ""
 }
 
-func handshakeWithExtension(ctx context.Context, conn net.Conn, zeroTime time.Time, address string, sni string,
-	extensions []utls.TLSExtension, logger model.Logger) *model.ArchivalTLSOrQUICHandshakeResult {
+func doHandshake(ctx context.Context, trace *measurexlite.Trace, conn net.Conn, zeroTime time.Time, address string, sni string,
+	echConfig []byte, logger model.Logger) *model.ArchivalTLSOrQUICHandshakeResult {
+	ctx, cancel := context.WithTimeout(ctx, 3*time.Second)
+	defer cancel()
 	tlsConfig := genTLSConfig(sni)
+	if len(echConfig) > 0 {
+		tlsConfig.EncryptedClientHelloConfigList = echConfig
```

Comment:
TODO: figure out if it's possible to pass in here something equivalent to GREASE